src/helpers/buildPaperslistByContact.js

```diff
@@ -3,23 +3,43 @@ import { isReferencedById } from 'cozy-client/dist/associations'
 import { CONTACTS_DOCTYPE } from 'src/doctypes'
 const { getFullname } = models.contact
 
+const DEFAULT_MAX_DISPLAY = 3
+
+/**
+ * @typedef {object} BuildPaperslistByContactParam
+ * @property {object[]} papersList - Array of IOCozyFile
+ * @property {object[]} contactsList - Array of cozy contact object
+ * @property {Paper[]} [papersDefinitions=[]] Array of Papers
+ * @property {string} [currentFileCategory=''] - Category of qualification
+ */
+
 /**
- * @param {IOCozyFile[]} papersList - An array of IOCozyFile
- * @param {object[]} contactsList - An array of cozy contact object
- * @returns {{ contact: string, papers: IOCozyFile[] }[]}
+ * @param {BuildPaperslistByContactParam} param
+ * @returns {{ contact: string, papers: {maxDisplay: number, list: IOCozyFile[]} }[]}
  */
-export const buildPaperslistByContact = (papersList, contactsList) => {
-  const result =
-    Array.isArray(contactsList) && Array.isArray(papersList)
-      ? contactsList.flatMap(contact => [
-          {
-            contact: getFullname(contact),
-            papers: papersList.filter(paper =>
-              isReferencedById(paper, CONTACTS_DOCTYPE, contact._id)
-            )
-          }
-        ])
-      : []
+export const buildPaperslistByContact = ({
+  papersList,
+  contactsList,
+  papersDefinitions = [],
+  currentFileCategory = ''
+}) => {
+  let result = []
+  if (Array.isArray(contactsList) && Array.isArray(papersList)) {
+    const currentDef = papersDefinitions.find(
+      paperDef => paperDef.label === currentFileCategory
+    )
+    result = contactsList.flatMap(contact => [
+      {
+        contact: getFullname(contact),
```

Comment:
If this 'contact' is then displayed, you shouldn't use getFullname but getDisplayName instead